### PR TITLE
Improve changelog check for pre-release branches

### PR DESCRIPTION
**Pull request by sanchariGr**

### CODE REVIEW

File: .github/workflows/continous-integration.yml [hunk 1]
```diff
@@ -194,9 +194,13 @@ jobs:
...
         uses: actions/checkout@v3
 
+      - name: Check for pre release branches
+        id: branch_name
+        run: echo "branch=$(echo ${GITHUB_REF##*/})" >> $GITHUB_ENV
+
       - name: Assert release includes all changelog entries
         # check changelog folder only when we create pull request preparing release
-        if: github.event_name == 'pull_request' && startsWith(github.head_ref, 'prepare-release')
+        if: github.event_name == 'pull_request' && startsWith(github.head_ref, 'prepare-release') && steps.branch_name.outputs.branch == r"\d*\.\d*\."
```
Comment (radovanZRasa) [inline .github/workflows/continous-integration.yml:203]: Let's add a comment which explains what does the regular expression matches.

**Review by radovanZRasa — COMMENTED** [review]
Let us use a more descriptive title for a PR.
Something like: Improve changelog check for pre-release branches

File: .github/workflows/continous-integration.yml [hunk 2]
```diff
@@ -194,9 +194,14 @@ jobs:
       - name: Checkout git repository 🕝
         uses: actions/checkout@v3
 
+      - name: Check for pre release branches
+        id: branch_name
+        run: echo "branch=$(echo ${GITHUB_REF##*/})" >> $GITHUB_ENV
```
Comment (ancalita) [inline .github/workflows/continous-integration.yml:199]: If we're storing this as env var, I'd capitalise the var name to `BRANCH`.

File: .github/workflows/continous-integration.yml [hunk 3]
```diff
@@ -194,9 +194,14 @@ jobs:
...
+
       - name: Assert release includes all changelog entries
         # check changelog folder only when we create pull request preparing release
-        if: github.event_name == 'pull_request' && startsWith(github.head_ref, 'prepare-release')
+        # check changelog folder only when the branch name does not have a pre-release version e.g.3.4.0a1
+        if: github.event_name == 'pull_request' && startsWith(github.head_ref, 'prepare-release') && steps.branch_name.outputs.branch != r"\d*\.\d*\.\w*\"
```
Comment (ancalita) [inline .github/workflows/continous-integration.yml:204]: Wouldn't it be easier to fetch the env var from env?
```suggestion
        if: github.event_name == 'pull_request' && startsWith(github.head_ref, 'prepare-release') && env.BRANCH != r"\d*\.\d*\.\w*\"
```

**Review by ancalita — COMMENTED** [review]
Looks good 👍🏻 Left some tiny suggestions, have you managed to test this locally with `act` tool?

**Review by radovanZRasa — APPROVED** [review]

**Comment by sanchariGr** [comment]
Forgot about this PR but found better solution here https://github.com/RasaHQ/rasa/pull/12442. Closing this.

**Pull request closed without merging**